src/v0/util/index.js

```diff
@@ -2201,6 +2201,26 @@ const combineBatchRequestsWithSameJobIds = (inputBatches) => {
   return combineBatches(combineBatches(inputBatches));
 };
 
+/**
+ * This function validates the event and return it as string.
+ * @param {*} isMandatory The event is a required field.
+ * @param {*} converToLowercase The event should be converted to lower-case.
+ * @returns {string} Event name converted to string.
+ */
+const validateEventAndLowerCaseConversion = (event, isMandatory, converToLowercase) => {
+  if (typeof event === 'object' || !isDefined(event)) {
+    throw new InstrumentationError('Event should not be a object, function or NaN and undefined');
+  }
+  if (!isMandatory) {
```

Comment:
if (isMandatory && event === 0 ) {
  throw new InstrumentationError('Event is a required field');
}

return converToLowercase ? event.toString().toLowerCase() : event.toString();

// restructure like this helps to reduce conditions

Reply:
Not thorwing error if 0, as some destinations take integer strings and "0" will be valid